cdap-common/bin/functions.sh

```diff
@@ -710,6 +710,9 @@ cdap_start_java() {
   if [[ ${CDAP_SERVICE} == master ]]; then
     # Determine SPARK_HOME
     cdap_set_spark || logecho "Could not determine SPARK_HOME! Spark support unavailable!"
+    if [[ -n ${SPARK_COMPAT} ]]; then
+      __defines+=" -Dapp.program.spark.compat=${SPARK_COMPAT}"
```

Comment:
this is so that the right value is used in the artifact directory setting in cdap-default.xml `/opt/cdap/master/artifacts;/opt/cdap/master/artifacts/${app.program.spark.compat}`